src/Chapter3.hs

```diff
@@ -374,6 +381,25 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Monster = MkMonster {
+  monsterHealth :: Int,
+  monsterAttack :: Int,
+  monsterGold :: Int
+}
+
+data Knight = MkKnight {
+  knightHealth :: Int,
+  knightAttack :: Int,
+  knightGold :: Int
+}
+
+fight :: Knight -> Monster -> Int
+fight MkKnight { knightHealth = kh, knightAttack = ka, knightGold = kg} 
+      MkMonster { monsterHealth = mh, monsterAttack = ma, monsterGold = mg}
+        | ka >= mh = mh + mg
+        | ma >= kh = -1
+        | otherwise = kg
```

Comment:
That's a great and correct solution! 👍🏻 You can also use a Haskell feature called `RecordWildCards` to reduce some code, as the field names are unambiguous. I wrote a blog post about feature, if you are interested 🙂 

* https://kodimensional.dev/recordwildcards